### DISCUSSION TITLE
Style customization in the PDF generation process

### DISCUSSION THREAD

**Discussion by yl5682** [thegetty/quire#290] [Development]
Dear community,

Long story short I am using the PDF generation feature to produce PDF for the Quire project. For the most part it works flawlessly. However there are a few issues that I hope to resolve.

**1. Break long strings in an URL in the Notes section.**
If you notice in the screenshot, the long URLs won't break because of the current word-break setting.
<img width="474" alt="Screen Shot 2022-04-22 at 6 06 27 AM" src="https://user-images.githubusercontent.com/7308403/164559701-91075e32-de3a-421b-923c-f7d6893d74e9.png">

Knowing that the `print.scss` is responsible for the style of the PDF, I added this:
```
  * {
    word-break: break-all;
    word-wrap: break-word;
  }
```

And the result looks like this:
<img width="345" alt="Screen Shot 2022-04-22 at 5 56 28 AM" src="https://user-images.githubusercontent.com/7308403/164559716-356b5318-042a-48a8-938d-db9c28a54a68.png">

The challenge is that it is breaking everything (as expected). So I wonder if there is a predefined selector that I can use to target only the Notes section (or footnote, if it is called that way)? Any recommendation would be great!

**2. Vertically center an image on a page**
My second question is also CSS related: Is there any way to vertically center the entire image/attribution section on the page for an image object? Please see the screenshot for when it is aligned at the top (desired layout is vertical center alignment).
<img width="501" alt="Screen Shot 2022-04-22 at 6 15 57 AM" src="https://user-images.githubusercontent.com/7308403/164560516-2a6f1c65-df3f-453f-a4c6-e604060bf611.png">

**3. Publisher information page**
Does the default PDF template/generator produce a publisher's information page? Currently I am not able to generate the PDF and get a page that has information about the publisher. A similar page that I am looking for may contain information such as:

<img width="303" alt="Screen Shot 2022-04-22 at 6 20 29 AM" src="https://user-images.githubusercontent.com/7308403/164561000-0a36dac8-beb5-4f73-ba98-7aff73d44505.png">

in this sample publication and I cite: https://highpoint-editions.netlify.app/downloads/mia-highpoint-editions.pdf

Thank you so much for the community's support!

**Comment by Erin-Cecele** [comment]
Hi @yl5682 to tackle your first question: 

You can see this bit of code in the `layouts/default/source/css/print.scss`: https://github.com/thegetty/quire/blob/5b533e0c6ff89ba8c9ff1e94a9d1c5de2524512e/themes/default/source/css/print.scss#L443-L463

The CSS selector there is targeting external links, but when Hugo switched Markdown processors away from Black Friday, the selector stopped working the way it should. The first thing you can try is changing the selector from `a[target=_blank]` to `a[href^="http"]`. 

Additionally you can try the following: 

To fix line breaks, they need to be written out in full Markdown link styling (brackets with the text as it appears, followed by parens for the URL). And the code for a zero-width non-breaking space (&#65279;) needs to be inserted following the character you want to break to the next line:

```
[https://www.brooklynmuseum.org/opencollection/objects/&#65279;152121](https://www.brooklynmuseum.org/opencollection/objects/152121)
```

Or, in some cases, parts of the URL need to be wrapped in HTML span tags:

```
[<span style="white-space:nowrap;">https://</span>www.brooklynmuseum.org/opencollection/objects/152121](https://www.brooklynmuseum.org/opencollection/objects/152121)
```

Let us know if these tips work. We'll be in touch to help respond to # 2 and # 3 shortly.

**Reply by yl5682** [reply]
I have forwarded this thread to the project team and they've found it very helpful. I wonder if you have had an update on the other issues? The team are planning to launch this project this Friday (sorry about the short notice). If you have some newly discovered pointers I am sure they will help the team a lot! Thank you @Erin-Cecele !

**Reply by geealbers** [reply]
Hi @yl5682, for the third question about the publisher information page, unfortunately no, Quire doesn't currently generate this automatically. It's something we hope to do something about in the future, but for now it has to be manually made, which I realize frustratingly goes against Quire's normal practice of only including content in one place and then using the templates to display that content as and where needed.

It might be useful to check out how I've been handling it for Getty books. Here's a repo of one of our recent titles: https://github.com/thegetty/rococo. In the content directory, you'll see both a [copyright.md](https://github.com/thegetty/rococo/blob/master/content/copyright.md) file for the pdf and epub output, and a very similar [about.md](https://github.com/thegetty/rococo/blob/master/content/about.md) file for the online site. I then usually add [some custom.css](https://github.com/thegetty/rococo/blob/1c19032fef8310d7a4e21c80eb03a490ad729a98/static/css/custom.css#L182-L228) to get the styles looking right. Hope this helps.

**Reply by geealbers** [reply]
Hi again @yl5682, this is the only solution I've found so far for your question about centering the images. It's working on my end, though I feel like there should be a better way. Let me know if it works for you.

Starting on line 266 of `themes/default/source/css/print.scss` replace this:

```scss
.quire-entry__image {
  .quire-entry__image__group-container {
    figure {
      page: entry-image;
    }
  }
}
```

with this:

```scss
$print-entry-image-height: $print-height - $print-top-margin - $print-bottom-margin;
.quire-entry__image {
  .quire-entry__image__group-container {
    figure {
      page: entry-image;
      height: $print-entry-image-height;
    }
  }
}
```
